### DISCUSSION TITLE
Sidecar Deploy error - Segmentation fault

### DISCUSSION THREAD

**Discussion by pepeloper** [stefanzweifel/sidecar-browsershot#65] [Q&A]
Hi, I would love to have some help here because I have this following error and I can't trace it down to see what is going on.

I have configured sidecar but when I run the command `php artisan sidecar:deploy` I got the following error

```
[Sidecar] Deploying Wnx\SidecarBrowsershot\Functions\BrowsershotFunction to Lambda as `sc-gestion-eventos-local-browsershot`.
          ↳ Environment: local
          ↳ Architecture: arm64
          ↳ Package Type: Zip
          ↳ Runtime: nodejs14.x
          ↳ Creating new lambda function.
          ↳ Packaging files for deployment.
[1]    57721 segmentation fault  php artisan sidecar:deploy
```

**Comment by stefanzweifel** [comment]
Does running `php artisan sidecar:deploy -vvv` show more information?

**Comment by pepeloper** [comment]
Thanks for the answer @stefanzweifel. Got the same output

```
[Sidecar] Deploying Wnx\SidecarBrowsershot\Functions\BrowsershotFunction to Lambda as `sc-gestion-eventos-local-browsershot`.
          ↳ Environment: local
          ↳ Architecture: arm64
          ↳ Package Type: Zip
          ↳ Runtime: nodejs14.x
          ↳ Creating new lambda function.
          ↳ Packaging files for deployment.
[1]    2544 segmentation fault  php artisan sidecar:deploy -vvv
```

I'll create an empty project from scratch and see if I can reproduce it and will get back here

**Comment by pepeloper** [comment]
Ok @stefanzweifel I managed to deploy a `HelloWorld` function working but the `BrowsershotFunction` always gives me that segmentation fault error

```
[Sidecar] Deploying App\Sidecar\HelloWorld to Lambda as `sc-gestion-eventos-local-sidecar-helloworld`.
          ↳ Environment: local
          ↳ Architecture: arm64
          ↳ Package Type: Zip
          ↳ Runtime: nodejs14.x
          ↳ Function already exists, potentially updating code and configuration.
          ↳ Packaging files for deployment.
          ↳ Creating a new zip file.
          ↳ Zip file created at s3://sidecar-eu-west-3-1682674293/sidecar/001-2a3e280f8d98f015bfad664034a7a3b7.zip. (0MB)
          ↳ Function code and configuration updated.
[Sidecar] Deploying Wnx\SidecarBrowsershot\Functions\BrowsershotFunction to Lambda as `sc-gestion-eventos-local-browsershot`.
          ↳ Environment: local
          ↳ Architecture: arm64
          ↳ Package Type: Zip
          ↳ Runtime: nodejs14.x
          ↳ Creating new lambda function.
          ↳ Packaging files for deployment.
[1]    61884 segmentation fault  php artisan sidecar:deploy -vvv
```

I'm using php 8.2.4 and Laravel 10 if it helps

**Reply by pepeloper** [reply]
Here is the repo: https://github.com/pepeloper/segmentation-fault

**Reply by stefanzweifel** [reply]
Thanks for the update and repo. I will take a closer look at this as soon as I can.

**Reply by stefanzweifel** [reply]
@pepeloper I think this is an issue related to the "Architecture" setting in your app. It looks like you're using `arm64` which is not compatible with sidecar-browsershot.

Our test suite deploys the Lambda function to AWS and it works fine when using the default `x86_64`.
https://github.com/stefanzweifel/sidecar-browsershot/actions/runs/4832963067/jobs/8612404713#step:6:18

It seems we have to add a `sidecar-browsershot.architecture` config to mitigate this. I'm sure you're using `arm64` for a good reason in your app.

**Reply by pepeloper** [reply]
Hey @stefanzweifel thanks for the help. I have been checking the repo I shared and there I have the `x86_64` architecture and I'm having the same issue. Will keep trying to know what it's going on, maybe it's just related to the sidecar package itself

```
[Sidecar] Deploying Wnx\SidecarBrowsershot\Functions\BrowsershotFunction to Lambda as `sc-pepeloper-local-browsershot`.
          ↳ Environment: local
          ↳ Architecture: x86_64
          ↳ Package Type: Zip
          ↳ Runtime: nodejs14.x
          ↳ Creating new lambda function.
          ↳ Packaging files for deployment.
[1]    99456 segmentation fault  php artisan sidecar:deploy
```

**Reply by stefanzweifel** [reply]
Quick update here. 
I update the test suite to also test `arm64`-deployments. https://github.com/stefanzweifel/sidecar-browsershot/pull/68

They don't work. I've released an update that always deploys sidecar-browsershot using `x86_64`. https://github.com/stefanzweifel/sidecar-browsershot/pull/67
